cachito/workers/pkg_managers/gomod.py

```diff
@@ -260,6 +245,59 @@ def resolve_gomod(app_source_path, request, dep_replacements=None, git_dir_path=
         return {"module": module, "module_deps": module_level_deps, "packages": packages}
 
 
+def _parse_name_and_version(list_deps_line: str, module_version: str) -> Tuple[str, str]:
+    # line example: pkg.io/foo/bar pkg.io/foo/var v1.0.0 => pkg.io/foo/bar v1.0.1
+    parts = list_deps_line.split(" ")
+
+    # As far as we know, the possible forms of the dependency line are as follows:
+    # | len | semantics                                              |
+    # |-----|--------------------------------------------------------|
+    # |   2 | <package> <module>                                     |
+    # |   3 | <package> <module> <version>                           |
+    # |   4 | <package> <module> => <./local/path>                   |
+    # |   5 | <package> <module> => <module> <version>               |
```

Comment:
@athos-ribeiro, how is it possible that we've never encountered this case? Here are some examples from processing kubernetes:
```
go.etcd.io/etcd/pkg/pathutil go.etcd.io/etcd => go.etcd.io/etcd v0.5.0-alpha.5.0.20200910180754-dd1b699fc489
github.com/coreos/go-semver/semver github.com/coreos/go-semver => github.com/coreos/go-semver v0.3.0
google.golang.org/protobuf/internal/detrand google.golang.org/protobuf => google.golang.org/protobuf v1.25.0
golang.org/x/net/context golang.org/x/net => golang.org/x/net v0.0.0-20201110031124-69a78807bb2b
```
And hundreds of other cases.

Reply:
Could this be a change from something newer than go `1.12` or `1.13`, when we started getting this information?

It would be nice to find some docs for these formates so we can link it here. I could not find any back then.

Reply:
There is some info in `go help list`, but the format for `.Module` is not described in detail. This is the relevant part:
```
The default output is to print the module path and then
information about the version and replacement if any.
For example, 'go list -m all' might print:

    my/main/module
    golang.org/x/text v0.3.0 => /tmp/text
    rsc.io/pdf v0.1.1
```
For what it's worth, when there is a replacement, the format seems to match the `ReplaceSpec` here: https://golang.org/ref/mod#go-mod-file-replace

Reply:
Added a comment